src/test/java/homeWork23/BookCartApiTests.java

```diff
@@ -0,0 +1,148 @@
+import static homeWork23.utils.DataGenerator.randomIntId;
+import static homeWork23.utils.TestData.*;
+import static org.testng.Assert.*;
+
+
+public class BookCartApiTests {
+
+    @Test
+    public void createUserValidData() {
+        Response response = UserService.createUser(randomUserData());
+        assertEquals(response.getStatusCode(), HttpStatusCode.OK.getCode());
+    }
+
+    @Test
+    public void createUserEmptyRequiredValues() {
+
+        UserDTO userDTO = emptyUserData();
+
+        Response response = UserService.createUser(userDTO);
+        assertEquals(response.getStatusCode(), HttpStatusCode.BAD_REQUEST.getCode());
+    }
+
+    @Test
+    public void loginValidData() {
+
+        UserDTO userDTO = randomUserData();
+        UserService.createUser(userDTO);
+
+        Response response = LoginService.loginUser(userDTO);
+        assertEquals(response.getStatusCode(), HttpStatusCode.OK.getCode());
+
+        LoginResponseDTO loginResponse = response.as(LoginResponseDTO.class);
+        assertEquals(loginResponse.getUserDTO().getUsername(), userDTO.getUsername());
+        assertNotNull(loginResponse.getToken());
+    }
+
+    @Test
+    public void loginUnexistingUser() {
+
+        UserDTO userDTO = randomUserData();
+
+        Response response = LoginService.loginUser(userDTO);
+        assertEquals(HttpStatusCode.UNAUTHORIZED.getCode(), response.getStatusCode());
+    }
+
+    @Test
+    public void addToCartValidData() {
+
+        UserDTO userDTO = randomUserData();
+        UserService.createUser(userDTO);
+
+        Integer userId = getCreatedUserId(LoginService.loginUser(userDTO));
+        Integer bookId = BookService.getBookId();
+
+        Response response = ShoppingCartService.addToCart(userId, bookId);
+        assertEquals(response.getStatusCode(), HttpStatusCode.OK.getCode());
+
+        String responseBody = response.getBody().asString();
+        assertTrue(responseBody.contains("1"));
+    }
+
+    @Test
+    public void addToCartUnexistingUser() {
+
+        Integer bookId = BookService.getBookId();
+
+        Response response = ShoppingCartService.addToCart(randomIntId(), bookId);
+
+        assertEquals(response.getStatusCode(), HttpStatusCode.NOT_FOUND.getCode());
+    }
+
+
+    @Test
+    public void getCart() {
+
+        UserDTO userDTO = randomUserData();
+        UserService.createUser(userDTO);
+
+        Integer userId = getCreatedUserId(LoginService.loginUser(userDTO));
+        Integer bookId = BookService.getBookId();
+
+        ShoppingCartService.addToCart(userId, bookId);
+        Response response = ShoppingCartService.getCart(userId);
+        assertEquals(HttpStatusCode.OK.getCode(), response.getStatusCode());
+
+        List<ShoppingCartResponseDTO> cart = response.jsonPath().getList("", ShoppingCartResponseDTO.class);
+        assertNotNull(cart);
+        assertTrue(containsBookIdInCart(cart, bookId));
```

Comment:
suppose that check can be extended as returned response should be next:
```
[
  {
    "book": {
      "bookId": 0,
      "title": "string",
      "author": "string",
      "category": "string",
      "price": 0,
      "coverFileName": "string"
    },
    "quantity": 0
  }
]
```

Reply:
i extended asserts, now i check all response fields comparing getCart and GetBooks responses and that i have one book in cart only